internal/infrastructure/kubernetes/utils/utils.go

```diff
@@ -30,3 +32,9 @@ func ExpectedServiceSpec(serviceType *egcfgv1a1.ServiceType) corev1.ServiceSpec
 	}
 	return serviceSpec
 }
+
+// CompareSvc Only compare the selector and ports(not include nodePort) in case user have modified for some scene.
+func CompareSvc(newSvc, originalSvc *corev1.Service) bool {
+	return cmp.Equal(newSvc.Spec.Selector, originalSvc.Spec.Selector) &&
```

Comment:
the infra layer also sets `Type` and `SessionAffinity`
https://github.com/envoyproxy/gateway/blob/690a045b4c54bd7a9acbdd4544f8575be99e2e8a/internal/infrastructure/kubernetes/utils/utils.go#L23

Reply:
something for the future, we should find a sane way to ensure we check what we set

Reply:
is there any need to update svc type rather than follow user's setting?

Reply:
added some comments above, `Patch` is fine as long as its easy to mantain